File: app/Dummy/c/src/core/chpass.c
// Copyright 05-Sept-2018 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

#include "core/chpass.h"
#include "dmc/cgi.h"

// 'rq' is Map[Js]
void chpass_process(Map *rq) {
  CGI_GET_STR(user, rq, "user");
  CGI_GET_STR(pass, rq, "pass");
  CGI_GET_STR(new_pass, rq, "newPass");
  cgi_change_pass(user, pass, new_pass);
  free(user);
  free(pass);
  free(new_pass);
}

File: app/Quotes2/h/src/Core/Chpass.hs
-- Copyright 27-Nov-2018 ºDeme
-- GNU General Public License - V3 <http://www.gnu.org/licenses/>

-- | Change password page

module Core.Chpass (process) where

import qualified Dm.Cgi as Cgi
import Dm.Cgi (Cgi)
import qualified Dm.Js as Js
import Dm.Js (JSValue)

process :: Cgi -> [(String, JSValue)] -> IO ()
process cgi rq =
  let user = Cgi.get rq Js.rString "user"
      pass = Cgi.get rq Js.rString "pass"
      newPass = Cgi.get rq Js.rString "newPass"
  in  Cgi.changePass cgi user pass newPass
